core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -231,6 +233,19 @@ private[spark] class EventLoggingListener(
     }
   }
 
+
+  private def redactEvent(event: SparkListenerEnvironmentUpdate): SparkListenerEnvironmentUpdate = {
```

Comment:
The downside is that with this choice you have more complicated tests, and you have to do redaction at every place where this information might be written (which at the moment is just two - the event logger and the UI).

JsonProtocol is not really a choice because it doesn't cover the UI.

> A user could write a custom listener that listened to the environment updates and did something useful with them. And, I didn't want to impose redaction on them.

That's the only argument I can buy, and said user has different ways to get that information.